ims/einstein/ceph.py

```diff
@@ -230,34 +231,46 @@ def get_parent_info(self, img_id):
             # Should be changed to special exception
             raise file_system_exceptions.ImageNotFoundException(img_id)
 
+    # Better Move to IET Driver as it is only using it
+    # Also will allow to decouple as this doesnt use rbd client
     @log
     def map(self, ceph_img_name):
-        command = "echo {0} | sudo -S rbd --keyring {1} --id {2} map {3}/{4}".format(
-            self.password, self.keyring, self.rid, self.pool, ceph_img_name)
-        p = subprocess.Popen(command, shell=True,
-                             stderr=subprocess.STDOUT, stdout=subprocess.PIPE)
-        output, err = p.communicate()
+        # command = "echo {0} | sudo -S rbd --keyring {1} --id {2} map {3}/{4}".format(
+        #     self.password, self.keyring, self.rid, self.pool, ceph_img_name)
+        # p = subprocess.Popen(command, shell=True,
+        #                      stderr=subprocess.STDOUT, stdout=subprocess.PIPE)
+        # output, err = p.communicate()
         # output = sh.rbd.map(ceph_img_name, keyring=self.keyring, id=self.rid,
         #            pool=self.pool)
-        if p.returncode == 0:
-            if output.find("sudo") != -1:
-                return output.split(":")[1].strip()
-            else:
+        try:
+            with sh.sudo:
+                output = sh.rbd.map(ceph_img_name, keyring=self.keyring,
+                                    id=self.rid,
+                                    pool=self.pool)
                 return output.strip()
```

Comment:
Also I can write another minimalistic wrapper that uses subprocess without shell=True to remove redundancy and give the same feel like sh.